core/src/org/javarosa/core/model/instance/TreeReference.java

```diff
@@ -252,25 +335,32 @@ public TreeReference getParentRef() {
         }
     }
 
-    //return a new reference that is this reference anchored to a passed-in parent reference
-    //if this reference is absolute, return self
-    //if this ref has 'parent' steps (..), it can only be anchored if the parent ref is a relative ref consisting only of other 'parent' steps
-    //return null in these invalid situations
-    public TreeReference parent(TreeReference parentRef) {
+    /**
+     * Join this reference with the base reference argument.
+     *
+     * @param baseRef an absolute reference or a relative reference with only
+     * '../'s
+     * @return a join of this reference with the base reference argument.
+     * Returns a clone of this reference if it is absolute, and null if this
+     * reference has '../'s but baseRef argument a non-empty relative reference.
+     */
+    public TreeReference parent(TreeReference baseRef) {
         if (isAbsolute()) {
-            return this;
+            return this.clone();
```

Comment:
We generally only Clone() when we modify, for performance reasons. This will work fine, I think, but not sure it's necessary.